Realm/ObjectStore/shared_realm.cpp

```diff
@@ -0,0 +1,430 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "shared_realm.hpp"
+
+#include "external_commit_helper.hpp"
+#include "realm_delegate.hpp"
+#include "transact_log_handler.hpp"
+
+#include <realm/commit_log.hpp>
+#include <realm/group_shared.hpp>
+
+#include <mutex>
+
+using namespace realm;
+
+RealmCache Realm::s_global_cache;
+
+Realm::Config::Config(const Config& c)
+: path(c.path)
+, read_only(c.read_only)
+, in_memory(c.in_memory)
+, cache(c.cache)
+, encryption_key(c.encryption_key)
+, schema_version(c.schema_version)
+, migration_function(c.migration_function)
+{
+    if (c.schema) {
+        schema = std::make_unique<Schema>(*c.schema);
+    }
+}
+
+Realm::Config& Realm::Config::operator=(realm::Realm::Config const& c)
+{
+    if (&c != this) {
+        *this = Config(c);
```

Comment:
This confused me at first. It's delegating copy assignment to the copy constructor and move assignment. Interesting technique!